prov/gni/src/gnix_vc.c

```diff
@@ -281,34 +281,34 @@ static void __gnix_vc_pack_conn_req(char *sbuf,
 	      sizeof(gni_mem_handle_t);
 	assert(len <= GNIX_CM_NIC_MAX_MSG_SIZE);
 
-	memcpy(cptr, &rtype, sizeof(rtype));
+	*((uint8_t *)cptr) = rtype;
 	cptr += sizeof(rtype);
```

Comment:
I also wanted to see more samples while testing these changes.  I did looked at many samples and noticed about a 35ns decrease in latency with these changes.  I will record more samples for these changes, if that’s still desired, but, I’m not sure how to measure network noise.  Any suggestions?
> On May 9, 2016, at 10:33 AM, James Swaro <notifications@github.com> wrote:
> 
> In prov/gni/src/gnix_vc.c <https://github.com/ofi-cray/libfabric-cray/pull/812#discussion_r62519676>:
> 
> > @@ -281,34 +281,34 @@ static void __gnix_vc_pack_conn_req(char *sbuf,
> >  	      sizeof(gni_mem_handle_t);
> >  	assert(len <= GNIX_CM_NIC_MAX_MSG_SIZE);
> >  
> > -	memcpy(cptr, &rtype, sizeof(rtype));
> > +	*((uint8_t *)cptr) = rtype;
> >  	cptr += sizeof(rtype);
> In general, I'd like to see more samples of this analysis. If there was only one measurement (before and after), I'd like to see more measurements. What was the difference in the maximum, minimum and average times between multiple runs? How can you prove that the difference in time isn't attributed to network noise?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/ofi-cray/libfabric-cray/pull/812/files/abc41c6cb76334b1f6e1fdc1a24af7a70c343be6#r62519676>